### PR TITLE
Rest params for function declarations

### PR DESCRIPTION
**Pull request by robertmaloney**
Part of #151 

### TODO
* [ ] tests for arrow functions
* [ ] rebase

### CODE REVIEW

File: src/lib/js/value.rs [hunk 1]
```diff
@@ -592,7 +592,15 @@ impl Display for ValueData {
             ValueData::Function(ref v) => match *v.borrow() {
                 Function::NativeFunc(_) => write!(f, "function() {{ [native code] }}"),
                 Function::RegularFunc(ref rf) => {
-                    write!(f, "function({}){}", rf.args.join(", "), rf.expr)
+                    write!(f, "function(")?;
+                    let last_index = rf.args.len() - 1;
+                    for (index, arg) in rf.args.iter().enumerate() {
```
Comment (IovoslavIovchev) [inline src/lib/js/value.rs]: This loop can be rewritten using `.iter().join()`.

Comment (robertmaloney) [reply src/lib/js/value.rs]: An attempt was made, but I'm having trouble converting `rf.args` into a format that supports `join`.

**Comment by jasonwilliams** [comment]
@robertmaloney this is great, do you know why the tests are failing?

**Comment by jasonwilliams** [comment]
Fixed test,
Looks like this doesn't handle spread syntax as an argument to a function, looking into that

**Comment by jasonwilliams** [comment]
This has already been implemented
https://github.com/jasonwilliams/boa/blob/master/src/lib/exec.rs#L823-L848

Thanks anyway @robertmaloney !

**Comment by robertmaloney** [comment]
Happy to see it taken care of, thanks for closing!

**Pull request closed without merging**